File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Arrays.sort(nums);

        int l = 0;
        int r = 1;
        boolean check = false;
        while(l < nums.length && r < nums.length && !check){
            if(nums[l] == nums[r]){
                check = true;
            }else{
                l++;
                r++;                
            }
        }
        return check;
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        while(l <= r):
            if(nums[l] == target):
                return l
            if(nums[r] == target):
                return r
            
            if(nums[int((r+l)/2)] < target):
                l = int((r+l)/2) + 1
            elif(nums[int((r+l)/2)] > target):
                r = int((r+l)/2) - 1
            else:
                return int((r+l)/2)

        return -1
            